contracts/exchange/test/match_orders.ts

```diff
@@ -1270,37 +1228,261 @@ describe('matchOrders', () => {
+                    leftMakerFee: TokenType.ERC721B,
+                    rightMakerFee: TokenType.ERC721A,
+                    leftTakerFee: TokenType.ERC20A,
+                    rightTakerFee: TokenType.ERC20A,
+                },
+                {
+                    description: 'Double-spend by trying to pay maker fees with sold ERC721 token (fail).',
+                    leftMaker: TokenType.ERC721A,
+                    rightMaker: TokenType.ERC721B,
+                    leftMakerFee: TokenType.ERC721A,
+                    rightMakerFee: TokenType.ERC721A,
+                    leftTakerFee: TokenType.ERC20A,
+                    rightTakerFee: TokenType.ERC20A,
+                    shouldFail: true,
+                },
+            ];
+
+            let erc721TokenIdsByOwner: ERC721TokenIdsByOwner;
+            let nameToERC20Tokens: { [name: string]: string };
+            let nameToERC721Tokens: { [name: string]: string };
+
+            function getAssetData(tokenType: TokenType, ownerAddress: string): string {
+                const encodeERC20AssetData = assetDataUtils.encodeERC20AssetData;
+                const encodeERC721AssetData = assetDataUtils.encodeERC721AssetData;
+                if (tokenType in nameToERC20Tokens) {
+                    const tokenAddress = nameToERC20Tokens[tokenType];
+                    return encodeERC20AssetData(tokenAddress);
+                }
+                if (tokenType in nameToERC721Tokens) {
+                    const tokenAddress = nameToERC721Tokens[tokenType];
+                    const tokenIdx = tokenType.charCodeAt(tokenType.length - 1) - 'A'.charCodeAt(0);
+                    const tokenId = erc721TokenIdsByOwner[ownerAddress][tokenAddress][tokenIdx];
+                    return encodeERC721AssetData(nameToERC721Tokens[tokenType], tokenId);
+                }
+                return '0x';
+            }
+
+            before(async () => {
+                erc721TokenIdsByOwner = await erc721Wrapper.getBalancesAsync();
+                nameToERC20Tokens = {
+                    ERC20A: erc20Tokens[0].address,
+                    ERC20B: erc20Tokens[1].address,
+                    ERC20C: erc20Tokens[0].address,
+                    ERC20D: erc20Tokens[1].address,
+                };
+                nameToERC721Tokens = {
+                    ERC721A: erc721Token.address,
+                    ERC721B: erc721Token.address,
+                    ERC721C: erc721Token.address,
+                    ERC721D: erc721Token.address,
+                };
+            });
+
+            for (const combo of feeAssetCombinations) {
+                const description = combo.description || JSON.stringify(combo);
+                it(description, async () => {
+                    // Create orders to match. For ERC20s, there will be a spread.
+                    const leftMakerAssetAmount = combo.leftMaker.startsWith('ERC20')
+                        ? Web3Wrapper.toBaseUnitAmount(15, 18)
```

Comment:
Is there anything special about these values or were they just picked at random?

Reply:
The values were taken from the existing ERC20 tests. I was wondering this too.